### PR TITLE
Test and Hinglish Classification

### PR DESCRIPTION
**Pull request by meghanabhange**

### CODE REVIEW

File: Alfaaz.py [hunk 1]
```diff
@@ -1,49 +1,78 @@
+import logging
```
Comment (NirantK) [inline Alfaaz.py:1]: Use isort and black please!

File: Alfaaz.py [hunk 2]
```diff
@@ -1,49 +1,78 @@
...
 def get_tweet(tweet_url: str, twitter_API):
     try:
-        tweet_id = re.findall("(?<=status\/)(.*)(?=\?s)", tweet_url)
+        tweet_id = re.findall("(?<=status\/)(.*)", tweet_url)
+        logger.info(f"tweet_id : {tweet_id}")
+        if tweet_id[-5:] == "?s=20":
+            tweet_id = tweet_id[:-5]
         tweet = twitter_API.get_status(tweet_id[0]).text
+        logger.info(f"Tweet Recieved : {tweet}")
     except:
```
Comment (NirantK) [inline Alfaaz.py:36]: Catch specific `Exception` and errors please?

File: Alfaaz.py [hunk 3]
```diff
@@ -1,49 +1,78 @@
...
-        st.e("Invalid tweet, please try again")
+        st.error("Invalid tweet, please try again")
```
Comment (NirantK) [inline Alfaaz.py:37]: This needs to be more descriptive than "Invalid Tweet". Maybe something like invalid twitter hash, auth failure, tweepy error or more.

File: Alfaaz.py [hunk 4]
```diff
@@ -1,49 +1,78 @@
...
 
 
 def get_tweet(tweet_url: str, twitter_API):
     try:
-        tweet_id = re.findall("(?<=status\/)(.*)(?=\?s)", tweet_url)
+        tweet_id = re.findall("(?<=status\/)(.*)", tweet_url)
+        logger.info(f"tweet_id : {tweet_id}")
+        if tweet_id[-5:] == "?s=20":
+            tweet_id = tweet_id[:-5]
         tweet = twitter_API.get_status(tweet_id[0]).text
```
Comment (NirantK) [inline Alfaaz.py:34]: I am guessing we should use 280 char, `full_text` field instead of `text`?

File: test.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import unittest
+
+from Alfaaz import get_tweet, get_twitter_api
+from sentiment_detection import HinglishSentiment
+
+
+class TestHinglishSentiement(unittest.TestCase):
```
Comment (NirantK) [inline test.py:7]: Maybe have 2-3 tweets per sentiment as test cases?

**Review by NirantK — CHANGES_REQUESTED** [review]

File: Alfaaz.py (hunk 1 above)
Comment (meghanabhange) [reply Alfaaz.py:1]: Did, I sort puts logging first :(

File: Alfaaz.py (hunk 2 above)
Comment (meghanabhange) [reply Alfaaz.py:36]: I thought using `logger.exception` was helping that. 
`logger.exception` logs levels at the `error` level and automagically adds the exception that caused it go in the except loop.

**Comment by NirantK** [comment]
@meghanabhange please resolve merge conflicts?

**Comment by meghanabhange** [comment]
Closing this :(

**Comment by NirantK** [comment]
What was this PR even about? Wow. We really need to treat with a little more respect @meghanabhange  :P

**Pull request closed without merging**